pipeline_dp/pipeline_backend.py

```diff
@@ -132,6 +132,14 @@ def flatten(self, col1, col2, stage_name: str):
         """
         pass
 
+    @abc.abstractmethod
+    def annotate(self, col, stage_name: str, **kwargs):
+        """
+        Returns:
+          The input collection after applying all the annotations in _annotators.
+        """
+        pass
```

Comment:
Nit:
`pass` -> `return col`  
and then
1. no need to have empty implementation in Spark/LocalBackend
2.remove @abstractmethod